pkg/sip/room.go

```diff
@@ -126,6 +132,21 @@ func (r *Room) subscribeTo(pub *lksdk.RemoteTrackPublication, rp *lksdk.RemotePa
 	r.subscribed.Break()
 }
 
+func (r *Room) getRoomLogger() logger.Logger {
+	select {
+	case <-r.ready.Watch():
+	case <-r.stopped.Watch():
+	}
```

Comment:
Is this really necessary? If it gets stuck here, it would be pretty hard to debug...

Could we instead either:
- Log `SID()` at a later time, so that logs for that call will eventually have it.
- ...or make SDK not block on `SID()` call (I'd prefer this one).

Reply:
I'd expect that removing the blocking on room.SID would break something as this doesn't seem like something that would be added randomly. It's also not trivial to audit project wide. The code would also still deadlock unless we still dispatch all log calls outside the participant callbacks as room.SID() attempts to take a lock that is already taken when the participant callback is called (these participant callbacks are overall not very safe right now)

It's about as easy to replace r.log with a version that includes the roomID once `ready` is true, when we know that room.SID() won't block. We can make logging the roomID best effort, but I'm afraid that it'll make the effort moot as we're debugging an issue that happens very shortly after join. I can still start there.